Primo.js

```diff
@@ -212,7 +213,13 @@ function importPNX(text) {
 	
 	var itemType = ZU.xpathText(doc, '//display/type');
 	if(!itemType) {
-		throw new Error('Could not locate item type');
+		if ( ZU.xpathText(doc, '//facets/rsrctype') ) {
+			var itemType = ZU.xpathText(doc, '//facets/rsrctype');
+		} else if ( ZU.xpathText(doc, '//search/rsrctype') ) {
+			var itemType = ZU.xpathText(doc, '//search/rsrctype');
+		} else {
```

Comment:
I don't have an example handy, but there are cases where the type identification failed for me. In fact you already account for those cases and throw an error. Can you elaborate on why throwing an error is better than using facets/search?